### PR TITLE
Backport style changes in scripted tests and Defaults.scala

### PR DESCRIPTION
**Pull request by eed3si9n**
This backports the scripted tests and Defaults.scala style changes to use `build.sbt` and `:=`.

The purpose is to catch mistakes earlier if there's any, and also to prepare for the `<<=` deprecation.

/review @dwijnand, @Duhemm

### CODE REVIEW

**Comment by dwijnand** [comment]
A little much to review, so a superficial LGTM - I'm sure it's all excellent.

**Comment by dwijnand** [comment]
Not merging in case you're doing your PRs in particular order and want to rebase (and/or squash) this first.

**Comment by eed3si9n** [comment]
This one is good to be merged given we agree on the general direction of `<<=` removal in sbt 1.0.

**Comment by dwijnand** [comment]
Ok cool.

Also I realise now that this is required for https://github.com/sbt/sbt/pull/2716, so merging this now.

File: main/src/main/scala/sbt/Opts.scala [hunk 1]
```diff
@@ -52,14 +52,14 @@ object DefaultOptions {
   def pluginResolvers(plugin: Boolean, snapshot: Boolean): Seq[Resolver] = {
     if (plugin && snapshot) Seq(Classpaths.typesafeSnapshots, Classpaths.sbtPluginSnapshots) else Nil
   }
-  def addResolvers: Setting[_] = Keys.resolvers <++= Keys.isSnapshot apply resolvers
-  def addPluginResolvers: Setting[_] = Keys.resolvers <++= (Keys.sbtPlugin, Keys.isSnapshot) apply pluginResolvers
```
Comment (dwijnand) [inline main/src/main/scala/sbt/Opts.scala:56]: removing this is a breaking change: https://github.com/scala/community-builds/pull/290

**Pull request merged**